### DISCUSSION TITLE
Can we control ingoings and outgoing edges for a given node port

### DISCUSSION THREAD

**Discussion by Patil-Mayur** [antvis/X6#1886] [Q&A] [answered]
Is there a way to allow only incoming edges to node's port 
Or
Allow only outgoing edges to node's port
with some color codes

also i checked the createEdge method triggered while drawing an edge from nopde.
if i return null it will use default edge instead of blocking edge creation.

**Comment by NewByVector** [comment, marked as answer]
Hopefully this [demo](https://codesandbox.io/s/limit-port-connection-ck62wf) will help you.

**Reply by Patil-Mayur** [reply]
I was more inclined towards getting port details here because wanted to define few rules based on port.
for eg: from one port red color is allowed, from other port green color outgoing edge allowed.
createEdge method does not give current selected port. It rather gives sourceCell details and we return shape.
Are you aware how could we acheive it?

or we need to fork it and add optional argument as selectedPort to createEdge method

**Reply by NewByVector** [reply]
You can use `validateEdge` to do this.

**Reply by Patil-Mayur** [reply]
validateEdge would validate the source and target connection. It would help to make construction decision.

eg: // drag and drop scenario

createEdge({selectedPort}) {
// provided source port select of a cell from multiple ports is valid
   if(selectedPort.id === 'negative-path') {
      return new Shape.Edge(`config with red color`);
   } else if (selectedPort.id === 'positive-path') {
     return new Shape.Edge(`config with green color`)
   }
  return new Shape.Edge(`config with default color`)
}

instead on selectedPort the control is on cell

**Reply by NewByVector** [reply]
I'm sorry, I misunderstood you.So let's first look at the definition of createEdge：
```ts
createEdge?: (
  this: Graph,
  args: {
    sourceCell: Cell
    sourceView: CellView
    sourceMagnet: Element
  },
) => Nilable<Edge> | void
```
We found That sourceMagnet can help you implement your feature.
